src/emoji/data.js

```diff
@@ -25,10 +21,19 @@ export const codeToEmojiMap = objectFromEntries<string, string>(
   }),
 );
 
-export const getFilteredEmojiNames = (
+export const getFilteredEmojis = (
   query: string,
   activeImageEmojiByName: $ReadOnly<{ [string]: ImageEmojiType }>,
-): string[] => {
+): $ReadOnlyArray<{| name: string, code: string, type: EmojiType |}> => {
   const names = [...unicodeEmojiNames, ...Object.keys(activeImageEmojiByName)];
-  return Array.from(new Set([...names.filter(x => x.indexOf(query) === 0).sort()]));
+  return Array.from(new Set([...names.filter(x => x.indexOf(query) === 0).sort()])).map(
+    emojiName => {
+      const isImageEmoji = activeImageEmojiByName[emojiName] !== undefined;
+      return {
+        name: emojiName,
+        code: isImageEmoji ? activeImageEmojiByName[emojiName].code : unicodeCodeByName[emojiName],
+        type: isImageEmoji ? 'image' : 'unicode',
```

Comment:
Let's call this property `emoji_type`, and the same in the downstream places it appears. That will help keep straight for the reader what kind of "type" this is.